envoyauth/response.go

```diff
@@ -394,3 +412,54 @@ func transformHeadersToEnvoy(input any) ([]*ext_core_v3.HeaderValueOption, error
 	}
 	return nil, fmt.Errorf("type assertion error, expected headers to be of type 'object' but got '%T'", input)
 }
+
+func transformToHTTPHeaderFormat(input interface{}) (http.Header, error) {
+	result := make(http.Header)
+
+	takeResponseHeaders := func(headers map[string]interface{}) error {
+		for key, value := range headers {
+			switch values := value.(type) {
```

Comment:
This seems to be a structural duplicate to the transformation for the envoy HttpOption. Do you want to unify this more?

Reply:
Yes indeed. Addressed.

Reply:
What I meant is that this switch statement in the for loop is essentially the same as in `makeEnvoyHeaderValueOptionsFromHeadersMap` so you could model this with a transformer + collector function and then the same for loop and structural swich could be used...

Reply:
Thanks for clarifying. I hope it is there now.